src/main/java/com/team4/testingsystem/services/impl/TestsServiceImpl.java

```diff
@@ -68,6 +72,15 @@ public Test getById(long id) {
         return testsRepository.getByStatuses(statuses);
     }
 
+    @Override
+    public void attachAssignedTests(List<UserDTO> users) {
```

Comment:
Let's return a map with id and test instead of modifying DTO on a service layer

Reply:
If I return a map from `attach...` instead of modifying DTO, other service will use DTO, or controller will have business-logic...

Reply:
Looks like an endless cycle 😳

Reply:
Can we have an object that will represent user and his test?
If so, we can return it from service, and controller can map it to dto on its layer only

Reply:
I can use smth like `Map<User, Optional<Test>>` for the service layer